sdx_controller/controllers/l2vpn_controller.py

```diff
@@ -243,7 +239,7 @@ def patch_connection(service_id, body=None):  # noqa: E501
         if remove_conn_code // 100 != 2:
             body, _ = connection_state_machine(body, ConnectionStateMachine.State.DOWN)
             db_instance.add_key_value_pair_to_db(
-                MongoCollections.CONNECTIONS, service_id, json.dumps(body)
+                MongoCollections.CONNECTIONS, service_id, body
```

Comment:
here we are basically updating the status, right? Wouldnt it be the case to leverage the function you introduced here? ie., `db_instance.update_field_in_json`

Reply:
Calling connection_state_machine() will ensure the transition is valid from state machine. So I think it'll still be necessary.